rmw_connextdds_common/src/common/rmw_impl_waitset_std.cpp

```diff
@@ -745,6 +796,23 @@ RMW_Connext_SubscriberStatusCondition::update_status_sample_lost(
     this->status_sample_lost_last.total_count;
 }
 
+void
+RMW_Connext_SubscriberStatusCondition::update_status_matched(
+  const DDS_SubscriptionMatchedStatus * const status)
+{
+  this->status_matched = *status;
+
+  if (status->current_count_change == 1) {
```

Comment:
@Barry-Xu-2018 
CC: @asorbini 

`current_count_change` will be either 1 or -1 always? otherwise, we will miss the event. as far as i see the specification, this could be 2 or -2 for example.

Reply:
I think the processing likes Fastdds.  
@asorbini please help to confirm this since I have no source codes.